app/server/publications/activities.js

```diff
@@ -18,7 +17,7 @@ ReactiveTable.publish(
       );
 
       // return mongo selector to fetch activities with matching resident IDs
-      return { residentIds: { $in: userVisibleActiveResidentIds } };
+      return Activities.find({ residentIds: { $in: userVisibleActiveResidentIds } });
```

Comment:
The idea is to do server-side pagination, so we avoid sending all the data to client.